The unadulterated version of this dataset was in XLSX format. In order for the Jupyter Notebook to open and thereby giving me access to the dataset, I converted it to a CSV. I then imported pandas to open the dataset.

In [2]:
import pandas as pd
Harley_Dataset = pd.read_csv('HarleyDavidsonSalesData.csv')

I then used the .head() to get a glimpse of this dataset:

In [3]:
Harley_Dataset.head()

Order Number                Product Name  Quantity  Price Payment Mode  \
0        101071  Harley-Davidson Street 750        30   7000   NetBanking   
1        101072  Harley-Davidson Street 750        34   7000         Cash   
2        101073  Harley-Davidson Street 750        41   7000  Credit card   
3        101074  Harley-Davidson Street 750        45   7000   NetBanking   
4        101075  Harley-Davidson Street 750        49   7000   NetBanking   

                            Store Name           City Country  Year  Month  \
0               RIDGES HARLEY-DAVIDSON            NYC     USA  2001      2   
1        SEVEN ISLANDS HARLEY-DAVIDSON          Reims  France  2001      5   
2              BANJARA HARLEY-DAVIDSON          Paris  France  2001      7   
3  TUSKER HARLEY-DAVIDSON LAVELLE ROAD       Pasadena     USA  2001      8   
4              CAPITAL HARLEY-DAVIDSON  San Francisco     USA  2001     10   

    Order Date  
0  24-Feb-2001  
1   7-May-2001  
2   1-Jul-2001  
3  25-Aug-2001  
4  10-Oct-2001

It looks good so far. Let's use the .info() to view details of the columns from this dataset:

In [4]:
Harley_Dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2823 entries, 0 to 2822
Data columns (total 11 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   Order Number  2823 non-null   int64 
 1   Product Name  2823 non-null   object
 2   Quantity      2823 non-null   int64 
 3   Price         2823 non-null   int64 
 4   Payment Mode  2823 non-null   object
 5   Store Name    2823 non-null   object
 6   City          2823 non-null   object
 7   Country       2823 non-null   object
 8   Year          2823 non-null   int64 
 9   Month         2823 non-null   int64 
 10  Order Date    2823 non-null   object
dtypes: int64(5), object(6)
memory usage: 242.7+ KB


The dataset contains 2,823 rows, 11 columns and 3 datatypes. I will now check for any null values:

In [5]:
Harley_Dataset[Harley_Dataset.columns[Harley_Dataset.isnull().any()]]

Empty DataFrame
Columns: []
Index: [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 60, 61, 62, 63, 64, 65, 66, 67, 68, 69, 70, 71, 72, 73, 74, 75, 76, 77, 78, 79, 80, 81, 82, 83, 84, 85, 86, 87, 88, 89, 90, 91, 92, 93, 94, 95, 96, 97, 98, 99, ...]

[2823 rows x 0 columns]

There are no columns containing null values. Great! I will now run one more inspection, just to be sure:

In [6]:
Harley_Dataset.isna().any().sum()

0

Phenomenal! As there are no null values in this dataset. I will now define the columns as a reference throughout the project:

0 Order Number: The order number of the product

1 Product Name: The name of Harley Davidson motorcycle

2 Quantity: The number of motorcycles included per order

3 Price: In US dollars, the cost per motorcycle

4 Payment: The method of payment

5 Store Name: The name of the store which receives the order

6 City: Name of the city of the store receiving the order

7 Country: Name of the country that receives the order

8 Year: Year in which the order took place

9 Month: Month in which the order took place

10 Order Date: The date on which the order took place, written as DD-MM-YYYY

Thus far, every column seems to provide valuable insight into the potential findings of this project. As such, I will keep every column for the time being. The dataset is now ready for the EDA portion of this project.

In [7]:
Harley_Dataset.to_csv('Harley_Dataset.csv', index = False)

In [8]:
Harley_Dataset['City'].unique()

array(['NYC', 'Reims', 'Paris', 'Pasadena', 'San Francisco', 'Burlingame',
       'Lille', 'Bergen', 'Melbourne', 'Newark', 'Bridgewater', 'Nantes',
       'Cambridge', 'Helsinki', 'Stavern', 'Allentown', 'Salzburg',
       'Chatswood', 'New Bedford', 'Liverpool', 'Madrid', 'Lule',
       'Singapore', 'South Brisbane', 'Philadelphia', 'Lyon', 'Vancouver',
       'Burbank', 'New Haven', 'Minato-ku', 'Torino', 'Boras',
       'Versailles', 'San Rafael', 'Nashua', 'Brickhaven', 'North Sydney',
       'Montreal', 'Osaka', 'White Plains', 'Kobenhavn', 'London',
       'Toulouse', 'Barcelona', 'Los Angeles', 'San Diego', 'Bruxelles',
       'Tsawassen', 'Boston', 'Cowes', 'Oulu', 'San Jose', 'Graz',
       'Makati City', 'Marseille', 'Koln', 'Gensve', 'Reggio Emilia',
       'Frankfurt', 'Espoo', 'Dublin', 'Manchester', 'Aaarhus',
       'Glendale', 'Sevilla', 'Brisbane', 'Strasbourg', 'Las Vegas',
       'Oslo', 'Bergamo', 'Glen Waverly', 'Munich', 'Charleroi'],
      dtype=object)

In [9]:
Harley_Dataset['Country'].unique()

array(['USA', 'France', 'Norway', 'Australia', 'Finland', 'Austria', 'UK',
       'Spain', 'Sweden', 'Singapore', 'Canada', 'Japan', 'Italy',
       'Denmark', 'Belgium', 'Philippines', 'Germany', 'Switzerland',
       'Ireland'], dtype=object)